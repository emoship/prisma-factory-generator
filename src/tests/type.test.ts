import { getDMMF } from '@prisma/sdk'
import {
  addModelFactoryParameterType,
  getModelFactoryParameterInterfaceProperties,
} from '../generator'
import { DMMF } from '@prisma/generator-helper'
import { Project } from 'ts-morph'

const datamodel = /* Prisma */ `
model User {
  id    Int    @id @default(autoincrement())
  email String @unique
  accessToken AccessToken?

  @@map(name: "users")
}

model AccessToken {
  id        Int      @id @default(autoincrement())
  user      User     @relation(fields: [userId], references: [id])
  userId    Int      @unique @map(name: "user_id")
  createdAt DateTime @default(now()) @map(name: "created_at")
  isActive  Boolean  @default(false)

  @@map("access_tokens")
}
`

let dmmf: DMMF.Document
let userModel: DMMF.Model
let accessTokenModel: DMMF.Model

beforeAll(async () => {
  dmmf = await getDMMF({ datamodel })
  const models = dmmf.datamodel.models
  userModel = models[0]
  accessTokenModel = models[1]
})

test('snapshot', () => {
  const project = new Project()
  const userSourceFile = project.createSourceFile('user')
  addModelFactoryParameterType(userSourceFile, userModel)
  expect(userSourceFile.getText()).toMatchSnapshot()

  const accessTokenSourceFile = project.createSourceFile('accessToken')
  addModelFactoryParameterType(accessTokenSourceFile, accessTokenModel)
  expect(accessTokenSourceFile.getText()).toMatchSnapshot()
})
