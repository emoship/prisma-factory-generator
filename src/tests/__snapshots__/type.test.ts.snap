// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
"type createUserArgs = RequiredParametersForUserCreation & Partial<Prisma.UserCreateArgs>
"
`;

exports[`snapshot 2`] = `
"type createAccessTokenArgs = RequiredParametersForAccessTokenCreation & Partial<Prisma.AccessTokenCreateArgs>
"
`;
